common/models/user.js

```diff
@@ -557,6 +571,9 @@ module.exports = function(User) {
       cb(err);
       return cb.promise;
     }
+    if (options.password && options.password.length > MAX_PASSWORD_LENGTH) {
+      this.validatePassword(options.password);
```

Comment:
We should always call `validatePasspord`. Please note the extended user model can override `validatePassword` to enforce more rules.